### ISSUE TITLE
Slim Dockerfile and correct behavior

### ISSUE THREAD

**Issue by rafaeldamasceno** [galnir/Master-Bot#521]
**Explain your suggestion**
Currently, the Dockerfile reclones the project and then copies the config.json and the json.sqlite files into the image. As is, this setup doesn't work for a first time run, since the json.sqlite file doesn't even exist. Furthermore, it stands to reason that if I have the Dockerfile to create a new image, I probably just cloned the project into my system which makes it very redundant to have the image build just clone it again.

To make the setup better, the created image should be slim and based on our local source files. For this, I propose the following Dockerfile changes:

- Do not clone the repository from source
- Remove development dependencies (git, vim, nano, etc.)
- Use the current source to get the project source files (it also makes it so that local changes can be tested)
- Copy the config AND option files (the latter is currently missing)
- Have instructions on how to setup a volume to persist guild commands

**Comment by galnir** [comment]
Merged #522 
A big thanks to @Bacon-Fixation and thank you @rafaeldamasceno for proposing these changes 👍 You're always welcome to propose changes or new features

**Comment by rafaeldamasceno** [comment]
I made some even further changes by myself but I wasn't as fast to create an MR :)

I've also added better a better .dockerignore and dependency steps, so that when you change something in the code, the cached dependency layers are used.

Before I create the MR, there's two things that I need to know:

1. Is there any reason why you use a Debian image? I've created some changes which further try to reduce the installed dependencies and it's based on a Node.js alpine image for further size reduction.
2. Is pm2 strictly necessary? What do we gain from using it inside a container?
